--- PhpStorm-111.19/bin/phpstorm.sh	2011-12-06 19:55:51.616124548 +0200
+++ PhpStorm-111.19/bin/phpstorm.sh	2011-12-06 21:11:56.432592623 +0200
@@ -63,9 +63,6 @@
   if [ -z "$JDK" ]; then
     echo "ERROR: cannot start WebIde."
     echo "No JDK found. Please validate either WEBIDE_JDK, JDK_HOME or JAVA_HOME environment variable points to valid JDK installation."
-    echo
-    echo "Press Enter to continue."
-    read IGNORE
     exit 1
   fi
 fi
@@ -74,8 +71,6 @@
 $JDK/bin/java -version 2> "$VERSION_LOG"
 grep 'OpenJDK' "$VERSION_LOG"
 OPEN_JDK=$?
-grep '64-Bit' "$VERSION_LOG"
-BITS=$?
 rm "$VERSION_LOG"
 if [ $OPEN_JDK -eq 0 ]; then
   echo "WARNING: You are launching IDE using OpenJDK Java runtime."
@@ -86,13 +81,6 @@
   echo "         please validate either WEBIDE_JDK, JDK_HOME, or JAVA_HOME environment variable points to valid Oracle (Sun) JDK installation."
   echo "         See http://ow.ly/6TuKQ for more info on switching default JDK"
   echo
-  echo "Press Enter to continue."
-  read IGNORE
-fi
-if [ $BITS -eq 0 ]; then
-  BITS="64"
-else
-  BITS=""
 fi
 
 #--------------------------------------------------------------------------
@@ -134,7 +122,7 @@
   # only extract properties (not VM options) from Info.plist
   INFO_PLIST_PARSER_OPTIONS=""
 else
-  [ "$BITS" == "64" ] && INFO_PLIST_PARSER_OPTIONS=" 64" || INFO_PLIST_PARSER_OPTIONS=" 32"
+  [ "$BITS" = "64" ] && INFO_PLIST_PARSER_OPTIONS=" 64" || INFO_PLIST_PARSER_OPTIONS=" 32"
 fi
 
 # in Mac OS X ./Contents/Info.plist describes all VM options & system properties
@@ -144,7 +132,12 @@
 
 REQUIRED_JVM_ARGS="-Xbootclasspath/a:../lib/boot.jar -Didea.no.jre.check=true -Didea.paths.selector=WebIde10 $IDE_PROPERTIES_PROPERTY $REQUIRED_JVM_ARGS"
 
-JVM_ARGS="$JVM_ARGS $REQUIRED_JVM_ARGS $MAC_VM_OPTIONS $AGENT"
+WEBIDE_CONFIG_HOME="${XDG_CONFIG_HOME:-$HOME/.config}/WebIde10"
+# if old dir does not exist, use new one
+if [ ! -d "$WEBIDE_CONFIG_HOME" ]; then
+	WEBIDE_CONFIG_HOME="${XDG_CONFIG_HOME:-$HOME/.config}/WebIde"
+fi
+JVM_ARGS="$JVM_ARGS $REQUIRED_JVM_ARGS -Dwebide.config.home=$WEBIDE_CONFIG_HOME $MAC_VM_OPTIONS $AGENT"
 
 CLASSPATH=../lib/bootstrap.jar
 CLASSPATH=$CLASSPATH:../lib/util.jar
--- PhpStorm-111.19/bin/idea.properties~	2011-11-28 11:44:59.000000000 +0200
+++ PhpStorm-111.19/bin/idea.properties	2011-12-06 19:58:00.925419184 +0200
@@ -12,22 +12,22 @@
 #---------------------------------------------------------------------
 # Uncomment this option if you want to customize path to WebIde config folder. Make sure you're using forward slashes
 #---------------------------------------------------------------------
-# idea.config.path=${user.home}/.WebIde/config
+idea.config.path=${webide.config.home}/config
 
 #---------------------------------------------------------------------
 # Uncomment this option if you want to customize path to WebIde system folder. Make sure you're using forward slashes
 #---------------------------------------------------------------------
-# idea.system.path=${user.home}/.WebIde/system
+idea.system.path=${webide.config.home}/system
 
 #---------------------------------------------------------------------
 # Uncomment this option if you want to customize path to user installed plugins folder. Make sure you're using forward slashes
 #---------------------------------------------------------------------
-# idea.plugins.path=${user.home}/.WebIde/config/plugins
+idea.plugins.path=${webide.config.home}/config/plugins
 
 #---------------------------------------------------------------------
 # Uncomment this option if you want to customize path to WebIde logs folder. Make sure you're using forward slashes
 #---------------------------------------------------------------------
-# idea.log.path=${user.home}/.WebIde/system/log
+idea.log.path=${webide.config.home}/system/log
 
 #---------------------------------------------------------------------
 # Maximum file size (kilobytes) IDE should provide code assistance for.
