--- PhpStorm-107.120/bin/PhpStorm.sh~	2011-05-19 06:21:07.000000000 +0300
+++ PhpStorm-107.120/bin/PhpStorm.sh	2011-06-01 11:53:18.701430989 +0300
@@ -27,19 +27,14 @@
   if [ -z "$WEBIDE_JDK" ]; then
     echo ERROR: cannot start WebIde.
     echo No JDK found to run WebIde. Please validate either WEBIDE_JDK, JDK_HOME or JAVA_HOME environment variable points to valid JDK installation.
-    echo
-    echo Press Enter to continue.
-    read IGNORE
     exit 1
   fi
 fi
 
-VERSION_LOG='/tmp/java.version.log'
+VERSION_LOG=$(mktemp java.version.XXXXXX 2>/dev/null || echo ${TMPDIR:-/tmp}/java.version.$$$RAND.log)
 $WEBIDE_JDK/bin/java -version 2> $VERSION_LOG
 grep 'OpenJDK' $VERSION_LOG
 OPEN_JDK=$?
-grep '64-Bit' $VERSION_LOG
-BITS=$?
 rm $VERSION_LOG
 if [ $OPEN_JDK -eq 0 ]; then
   echo WARNING: You are launching IDE using OpenJDK Java runtime
@@ -48,15 +43,8 @@
   echo
   echo NOTE:    If you have both Sun JDK and OpenJDK installed
   echo          please validate either WEBIDE_JDK or JDK_HOME environment variable points to valid Sun JDK installation
-  echo
-  echo Press Enter to continue.
-  read IGNORE
-fi
-if [ $BITS -eq 0 ]; then
-  BITS="64"
-else
-  BITS=""
 fi
+BITS=""
 
 #--------------------------------------------------------------------------
 #   Ensure the WEBIDE_HOME var for this script points to the
@@ -88,12 +76,13 @@
 
 # PROFILE
 if [ "utils$_run_closure5@1898c18" = "true" ]; then
- AGENT="-agentlib:yjpagent$BITS=disablej2ee,sessionname=webide"
+ AGENT="-agentlib:yjpagent=disablej2ee,sessionname=webide"
 fi
 
 REQUIRED_JVM_ARGS="-Xbootclasspath/a:../lib/boot.jar -Didea.no.jre.check=true $AGENT $WEBIDE_PROPERTIES_PROPERTY $REQUIRED_JVM_ARGS"
-JVM_ARGS=`tr '\n' ' ' < "$WEBIDE_VM_OPTIONS"`
-JVM_ARGS="$JVM_ARGS $REQUIRED_JVM_ARGS"
+JVM_ARGS=$(tr '\n' ' ' < "$WEBIDE_VM_OPTIONS")
+WEBIDE_CONFIG_HOME="${XDG_CONFIG_HOME:-$HOME/.config}/WebIde10"
+JVM_ARGS="$JVM_ARGS $REQUIRED_JVM_ARGS -Dwebide.config.home=$WEBIDE_CONFIG_HOME"
 
 CLASSPATH=../lib/bootstrap.jar
 CLASSPATH=$CLASSPATH:../lib/util.jar

--- PhpStorm-103.99/bin/idea.properties~	2011-02-14 11:44:54.000000000 +0200
+++ PhpStorm-103.99/bin/idea.properties	2011-03-11 07:48:34.638262763 +0200
@@ -10,13 +10,13 @@
 # Note for Windows users: please make sure you're using forward slashes. I.e. c:/idea/system
 
 # path to IDEA config folder. Make sure you're using forward slashes
-idea.config.path=${user.home}/.WebIde10/config
+idea.config.path=${webide.config.home}/config
 
 # path to IDEA system folder. Make sure you're using forward slashes
-idea.system.path=${user.home}/.WebIde10/system
+idea.system.path=${webide.config.home}/system
 
 # path to user installed plugins folder. Make sure you're using forward slashes
-idea.plugins.path=${user.home}/.WebIde10/config/plugins
+idea.plugins.path=${webide.config.home}/config/plugins
 
 #---------------------------------------------------------------------
 # Maximum file size (kilobytes) IDEA should provide code assistance for.
